docs/source/caching.rst

```diff
@@ -0,0 +1,270 @@
+
+* Solver information about the previously cached solve.
+* Release times information about when each package variant in the resolve was last released.
+* Variant states information about the state of a variant. For example, in the 'filesystem' repository type, the 'state' is the last modified date of the file associated with the variant (perhaps a package.py). If the state of any variant has changed from a cached resolve - eg. if a file has been modified - the cached resolve is discarded.
+
+Setup
+-----
+
+To enable memcached caching, you need to configure the :data:`memcached_uri` config variable. This variable accepts a list of memcached uri servers or None. Example with memcached running on localhost on its default port:
+
+.. code-block:: console
+
+   memcached_uri = ["127.0.0.1:11211"]
+
+This is the only parameter you need to configure to enable caching of the content and location of package file definitions and resolutions in Rez.
+
+Configuration
+-------------
+There are several variables that can be accessed to modify the default behavior of resolve caching:
+
+* :data:`resolve_caching`
+* :data:`cache_package_files`
+* :data:`cache_listdir`
+* :data:`resource_caching_maxsize`
+* :data:`memcached_package_file_min_compress_len`
+* :data:`memcached_context_file_min_compress_len`
+* :data:`memcached_listdir_min_compress_len`
+* :data:`memcached_resolve_min_compress_len`
+
+
+Validating operation
+--------------------------
+To print debugging information about memcached usage, you can temporarily declare the following variables in a terminal:
+
+.. code-block:: console
+
+   export REZ_DEBUG_MEMCACHE=1 (linux/macos bash)
+   $env:REZ_DEBUG_MEMCACHE=1 (powershell)
+
+or set :data:`debug_memcache` to True in your rezconfig.py.
+
+Cache invalidation
+----------------------
+Cache entries will automatically be invalidated when a newer package version is released that would change the result
+of an existing resolve.
+
+For example, let's say you are running rez-env with the package ``foo1+<2``, and originally, the only available
+``foo`` package version is ``1.0.0``, so the cached resolve points to ``1.0.0``. However, at some point afterwards
+you release a new version ``1.0.1``. The cache would invalidate for the request ``foo1+<2`` and the next resolve
+would correctly retrieve package version ``1.0.1``.
+
+
+Show stats from memcached server
+--------------------------------
+Rez provides a command-line tool :ref:`rez-memcache` for query the memcached server and obtaining status information and statistics.
+
+.. code-block:: console
+
+   $ rez-memcache
+
+   CACHE SERVER               UPTIME      HITS      MISSES  HIT RATIO  MEMORY  USED
+   ------------               ------      ----      ------  ---------  ------  ----
+   127.0.0.1:11211            20 hours    27690     5205    84%        119 Gb  10 Mb (0%)
+   central.example.com:11211  6.2 months  19145089  456     99%        64 Mb   1.9 Mb (2%)
+
+Benefits
```

Comment:
I would put both the benefits and downside right at the top, right under "Resolve Caching". I would also not make them sections.